src/src/com/microsoft/aad/adal/ApplicationReceiver.java

```diff
@@ -54,35 +58,48 @@
      * defined in your manifest.
      */
     @Override
-    public void onReceive(Context context, Intent intent) {
+    public void onReceive(Context context, Intent intent) 
+    {
         // Check if the application is install and belongs to the broker package
-        if (intent.getAction().equals(Intent.ACTION_PACKAGE_ADDED)) {
-            Logger.v(TAG, "Application install message is received");
-            if (intent != null && intent.getData() != null) {
-                Logger.v(TAG, "Installing:" + intent.getData().toString());
-                if (intent
-                        .getData()
-                        .toString()
-                        .equalsIgnoreCase(
-                                "package:" + AuthenticationConstants.Broker.AZURE_AUTHENTICATOR_APP_PACKAGE_NAME)) {
-                    Logger.v(TAG, "Message is related to the broker");
+        final String methodName = ":onReceive";
+        if (intent.getAction().equals(Intent.ACTION_PACKAGE_ADDED)) 
+        {
+            Logger.v(TAG + methodName, "Application install message is received");
+            if (intent != null && intent.getData() != null) 
+            {
+                Logger.v(TAG + methodName, "Installing:" + intent.getData().toString());
+                final String receivedInstalledPackageName = intent.getData().toString();
+                if (receivedInstalledPackageName.equalsIgnoreCase("package:" + AuthenticationConstants.Broker.AZURE_AUTHENTICATOR_APP_PACKAGE_NAME) ||
+                        receivedInstalledPackageName.equalsIgnoreCase("package:" + AuthenticationSettings.INSTANCE.getBrokerPackageName()))
```

Comment:
At some point, we need to move to the way we reference the azure_authenticator_app_package_name for CP too.